File: java_brain/src/main/java/org/netbrains/javabrain/neural/BiasNeuron.java
package org.netbrains.javabrain.neural;

public class BiasNeuron extends Neuron {
    public BiasNeuron(int layerIndex, int index, double value) {
        super(layerIndex, index);
        this.value = value;
    }

    @Override
    public double activate() {
        return value;
    }
}

File: cpp_brains/biasneuron.cpp
#include "biasneuron.h"

BiasNeuron::BiasNeuron(int layerIndex, double value)
    :Neuron{layerIndex, -1}
{
    m_value = value;
}

double BiasNeuron::activate()
{
    return m_value;
}
